File: 프로그래머스/2/42578. 의상/의상.py
from itertools import combinations

def solution(clothes):
    clothes_dict = {}
    for name, category in clothes:
        if category not in clothes_dict:
            clothes_dict[category] = [name]
            continue
        clothes_dict[category].append(name)
        
    answer = 1
    for category, names in clothes_dict.items():
        answer *= len(names) + 1
    return answer - 1

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {
        var answer = 1;
        
        var closet = new HashMap<String, ArrayList<String>>();
        for (var c: clothes) {
            var name = c[0]; var kind = c[1];
            
            if (!closet.containsKey(kind)) {
                var newWears = new ArrayList<String>();
                newWears.add(name);
                closet.put(kind, newWears);
            } else {
                var wears = closet.get(kind);
                wears.add(name);
            }
        }
        
        for (var kind: closet.keySet()) {
            var wears = closet.get(kind);
            answer *= wears.size() + 1;
        }
        answer -= 1;
        
        return answer;
    }
}